lib/clickhousex/http_client.ex

```diff
@@ -7,42 +7,54 @@ defmodule Clickhousex.HTTPClient do
 
   @req_headers [{"Content-Type", "text/plain"}]
 
-  def send(query, request, base_address, timeout, nil, _password, database) do
-    send_p(query, request, base_address, database, timeout: timeout, recv_timeout: timeout)
+  def send(query, request, base_address, timeout, nil, _password, database, _opts) do
+    send_p(query, request, base_address, database, [timeout: timeout, recv_timeout: timeout], [])
   end
 
-  def send(query, request, base_address, timeout, username, password, database) do
-    opts = [hackney: [basic_auth: {username, password}], timeout: timeout, recv_timeout: timeout]
-    send_p(query, request, base_address, database, opts)
+  def send(query, request, base_address, timeout, username, password, database, opts) do
+    async = Keyword.get(opts, :async, false)
+    async_callback = Keyword.get(opts, :async_callback)
+    async_opts = [async: async, async_callback: async_callback]
```

Comment:
```suggestion
    async_opts = Keyword.take(opts, [:async,:async_callback])
```

Reply:
if we use the "Keyword.take", it will cause the following error 
```

[error] GenServer #PID<0.1750.0> terminating
** (RuntimeError) connect raised CaseClauseError exception. The exception details are hidden, as they may contain sensitive data such as database credentials. You may set :show_sensitive_data_on_connection_error to true when starting your connection if you wish to see all of the details
    (clickhousex 0.4.0) lib/clickhousex/http_client.ex:86: Clickhousex.HTTPClient.do_post_http/5
    (clickhousex 0.4.0) lib/clickhousex/http_client.ex:45: Clickhousex.HTTPClient.send_p/6
    (clickhousex 0.4.0) lib/clickhousex/protocol.ex:44: Clickhousex.Protocol.connect/1
    (db_connection 2.4.3) lib/db_connection/connection.ex:82: DBConnection.Connection.connect/2
    (connection 1.1.0) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 4.1.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

Reply:
@Abdallahj94 I fixed the issue, it was because of the Keyword.take(), did not set the default value if we did not pass the opts, and our logic always depends on the async flag, So I handled it by fetching the async flag using `Keyword.get(async_opts, :async, false)` on do_post_http function, please review the last changes :)